core/src/main/scala/org/apache/spark/sql/delta/commands/optimize/ZOrderMetrics.scala

```diff
@@ -27,10 +27,11 @@ case class ZOrderFileStats(num: Long, size: Long)
 
 object ZOrderFileStats {
   def apply(v: Iterable[(Int, Long)]): ZOrderFileStats = {
-    v.foldLeft(ZOrderFileStats(0, 0)) { (a, b) =>
-      ZOrderFileStats(a.num + b._1, a.size + b._2)
+      val (is, ls) = v.foldLeft((0L, 0L)) {
+        case ((i0, l0), (i1, l1)) => (i0 + i1, l0 + l1)
+      }
+      ZOrderFileStats(is, ls)
```

Comment:
This change just uses a `Tuple2` instead of `ZorderFileStats`. From my understanding (please correct me if I'm wrong), Scala tuples and case classes are treated very similarly so this change wouldn't have any sort of performance impact.

Do you think the change is more readable, or is there another reason you feel this is necessary?

Reply:
Yeah, I know that tuple is similar to case classes and it makes little difference in terms of peformance. While writing this code, I wondered why it uses foldLeft instead of while loop and temporary variable( or recursive function with `@tailrec`). Is there any reason for this?  

Therefore, I keep foldLeft and remove ZOrderFileStats, which has little to do with summing up file numbers and sizes, for slight improvement of readability. 
If there is no concern, I will replace it with while loop for performance.

Reply:
`foldLeft` is tail recursive so I don't think a while loop would be performance improvement. Unless you think a while loop would be significantly more readable, I don't think this change is necessary. I personally don't find either version to be more readable than the other.